### PR TITLE
Add tests for match command

### PR DESCRIPTION
**Pull request by andreabio**
## Summary by Sourcery

Add comprehensive tests for the match functionality and wire the correct match parser into the CLI entrypoint.

Enhancements:
- Correct the CLI import to use the match_parser module when registering the match subcommand.

Tests:
- Add unit tests for match dosage calculation, haplotype handling, chromosome resolution, and overall match rate computation, including edge cases such as missing data and no overlapping positions.
- Add integration-style tests for run_match reading VCF/tract inputs and writing expected TSV outputs for phased and unphased samples, extra tract columns, and malformed tract files.
- Add tests for the match subcommand parser to validate required arguments, defaults, and runner wiring.

### CODE REVIEW

**Comment by xin-huang** [comment]
@codex review

File: sstar/match.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+        tgt_chrom = _resolve_chrom(chrom, tgt_data)
+        src_chrom = _resolve_chrom(chrom, src_data)
+        tgt_gt = tgt_data[tgt_chrom]["GT"]
+        tgt_pos = np.asarray(tgt_data[tgt_chrom]["POS"])
+        src_gt = src_data[src_chrom]["GT"]
+        src_pos = np.asarray(src_data[src_chrom]["POS"])
+
+        tgt_region_pos = tgt_pos[(tgt_pos > start) & (tgt_pos <= end)]
+        src_region_pos = src_pos[(src_pos > start) & (src_pos <= end)]
```
[the remark this answers is not in the document]
Comment (andreabio) [reply sstar/match.py:250]: The interval logic is intentional because the tract coordinates are BED-style 0-based half-open [start, end), while the arrays being filtered contain 1-based VCF POS values. Therefore, a VCF position belongs to the BED interval when start < POS <= end. So no change implementations for this one.

File: tests/test_match.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                [[1, 1]],
+                [[1, 1]],
+            ]
+        )
+    )
+    pos = np.array([100, 200, 300])
+
+    assert match(tgt_gt, pos, 0, src_gt, pos, 0, [100, 200, 300], P=2) == pytest.approx(
+        (1.0 + 0.5 + 1.0) / 3
+    )
```
[the remark this answers is not in the document]
Comment (andreabio) [reply tests/test_match.py:139]: Accepted. I will add a separate test_match_all_non_callable_returns_na() test to cover the edge case where all queried positions are non-callable and match(...) should return "NA". I will keep the existing fully callable test_match() unchanged.

File: tests/test_match.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+def test_run_match(tmp_path):
+    vcf_file, tgt_ind_file, src_ind_file, tract_file, output_file = _write_match_input_files(
+        tmp_path, "21\t0\t250\tind1_1\n"
+    )
+
+    run_match(
+        vcf_file=str(vcf_file),
+        tgt_ind_file=str(tgt_ind_file),
+        src_ind_file=str(src_ind_file),
+        tract_file=str(tract_file),
```
[the remark this answers is not in the document]
Comment (andreabio) [reply tests/test_match.py:222]: Accepted. I will add a focused test for run_match(...) with a malformed tract file containing only three columns, and assert that it raises ValueError with the expected “tract file must contain at least four columns” message.

File: tests/test_match.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+def test_match_parser_accepts_required_args(tmp_path):
+    vcf_file, tgt_ind_file, src_ind_file, tract_file, output_file = _write_match_input_files(
+        tmp_path, "21\t0\t250\tind1_1\n"
+    )
+    parser = _sstar_cli_parser()
+
+    args = parser.parse_args(
+        [
+            "match",
+            "--vcf",
```
[the remark this answers is not in the document]
Comment (andreabio) [reply tests/test_match.py:376]: Accepted partially. I will extend the CLI parser test to assert that the match subcommand attaches a runner attribute and that it is callable. I will not add the monkeypatch invocation part, because replacing args.runner with a fake function would only test the fake function rather than the actual parser wiring.

**Comment by andreabio** [comment]
@sourcery-ai review

**Comment by andreabio** [comment]
@codex review

File: tests/test_match.py [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+
+def test_calc_match_pct():
+    assert calc_match_pct([0, 1, 2], [0, 1, 2], P=2) == 1.0
+    assert calc_match_pct([0], [1], P=2) == 0.5
+    assert calc_match_pct([0], [2], P=2) == 0.0
+    assert calc_match_pct([0, 1, 2], [0, 2, 2], P=2) == pytest.approx(
+        (1.0 + 0.5 + 1.0) / 3
+    )
+    assert calc_match_pct([0, -1], [0, 2], P=2) == 1.0
+    assert calc_match_pct([-1], [-1], P=2) == "NA"
```
[the remark this answers is not in the document]
Comment (andreabio) [reply tests/test_match.py:68]: Accepted partially. I added the empty input case because it matches the existing no callable sites behavior and returns "NA". I am leaving mismatched length validation out of this PR because the current production code does not explicitly define that behavior, and the normal match workflow constructs aligned dosage arrays from the same positions list. That validation could be handled separately if necessary.

**Comment by andreabio** [comment]
@sourcery-ai review

**Comment by andreabio** [comment]
@codex review

File: tests/test_match.py [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+def test_match_parser_accepts_required_args(tmp_path):
+    vcf_file, tgt_ind_file, src_ind_file, tract_file, output_file = _write_match_input_files(
+        tmp_path, "21\t0\t250\tind1_1\n"
+    )
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest="subparsers")
+    add_match_parser(subparsers)
+
+    args = parser.parse_args(
+        [
```
[the remark this answers is not in the document]
Comment (andreabio) [reply tests/test_match.py:374]: Accepted partially. I added a negative parser test asserting that `parser.parse_args(["match"])` raises `SystemExit` when required match arguments are missing. I kept the existing positive parser test with the current CLI argument names (`--vcf`, `--tgt`, `--src`, `--tract-file`, `--output`). I did not add the monkeypatch runner invocation in this PR because the test already verifies that `args.runner` exists and is callable, and invoking it would add extra mocking complexity without changing the current CLI contract.

**Comment by andreabio** [comment]
@sourcery-ai review

**Comment by andreabio** [comment]
@codex review

**Review by xin-huang — APPROVED** [review]
